src/main/java/com/karankumar/bookproject/backend/service/UserService.java

```diff
@@ -86,6 +87,15 @@ public void register(User user) throws UserAlreadyRegisteredException {
         authenticateUser(user);
     }
 
+    public User getCurrentUser() {
+        String username = SecurityContextHolder.getContext().getAuthentication().getName();
+        return userRepository.findByUsername(username).orElseThrow();
```

Comment:
How are we handling the exception? Shall we show a message etc on the UI?

Reply:
This should **never** happen. It would mean that your app granted access to some user that is actually not in the databse.

Reply:
Sure I agree it should never happen but how do we want to react when it happens (for whatever reasons)? Shall we pass the unhandled exception or we should return a proper error message?

Reply:
If I was a user, and I wanted to perform some action (create a custom shelf for example) and this method had thrown an exception... it would be great to know that my action was not successful. Maybe we could display a message that an unexpected error has occured and log it (later on even send an email to devs or whatever). The same could apply for other unexpected errors (that should never happen), e.g. predefined shelf does not exist.